styx-scheduler-service/src/main/java/com/spotify/styx/state/StateUtil.java

```diff
@@ -61,8 +61,11 @@ private StateUtil() {
                                                            Instant instant,
                                                            TimeoutConfig ttl) {
     return activeStates.parallelStream()
-        .filter(entry -> hasTimedOut(workflows.get(entry.workflowInstance().workflowId()), entry.runState(), instant,
```

Comment:
We should look into using static analysis to catch possible NPEs.

https://github.com/uber/NullAway
https://checkerframework.org/tutorial/webpages/get-started-cmd.html